### PR TITLE
communities: add permission checks for submission policy

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/CERNDocumentServer/cds-rdm/issues/158

CommunityListItem:
Screenshot 2024-08-07 at 16 29 48

Error that is shown when clicking on "Submit for review"
Screenshot 2024-08-07 at 16 32 21

### CODE REVIEW

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunitySelectionModal/CommunityListItem.js [hunk 1]
```diff
@@ -45,6 +46,19 @@ export const CommunityListItem = ({ result, record, isInitialSubmission }) => {
...
         />
       )}
+      {!canSubmitRecord && (
+        <InvenioPopup
+          popupId="community-inclusion-info-popup"
+          size="small"
+          trigger={
+            <Icon className="mb-5" color="grey" name="question circle outline" />
+          }
+          ariaLabel={i18next.t("Community submission policy information")}
```
Comment (zzacharo) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunitySelectionModal/CommunityListItem.js:56]: See https://github.com/inveniosoftware/invenio-communities/pull/1189#discussion_r1717066634

File: tests/conftest.py [hunk 2]
```diff
@@ -990,6 +990,15 @@ def closed_review_minimal_community(minimal_community):
     return community
 
 
+@pytest.fixture()
+def closed_submission_minimal_community(minimal_community):
+    """Data for a minimal community that restricts record submission."""
+    community = deepcopy(minimal_community)
+    community["slug"] = "closed-submission-community"
+    community["access"]["submission_policy"] = "closed"
```
Comment (zzacharo) [inline tests/conftest.py:998]: it should be changed after the renaming proposed in https://github.com/inveniosoftware/invenio-communities/pull/1189

File: tests/resources/serializers/test_marcxml_serializer.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def updated_full_record(full_record_to_dict):
                 "access": {
                     "member_policy": "open",
                     "members_visibility": "public",
-                    "record_policy": "open",
+                    "submission_policy": "open",
```
Comment (zzacharo) [inline tests/resources/serializers/test_marcxml_serializer.py:29]: ditto renaming

File: tests/resources/serializers/test_marcxml_serializer.py [hunk 4]
```diff
@@ -48,7 +48,7 @@ def updated_full_record(full_record_to_dict):
...
```
Comment (zzacharo) [inline tests/resources/serializers/test_marcxml_serializer.py:51]: ditto renaming

File: tests/resources/test_resources_review.py [hunk 5]
```diff
@@ -224,3 +224,233 @@ def test_delete_no_review(
...
+
+
+def test_submission_closed_member(
+    client,
+    minimal_record,
+    closed_submission_community,
+    headers,
+    community_owner,
+):
+    """Test different users ability ro submit record with closed submission policy."""
```
Comment (zzacharo) [inline tests/resources/test_resources_review.py:281]:
```suggestion
    """Test different users ability to submit record with closed submission policy."""
```

maybe a bit more descriptive docstring for the specific case this tests validates so we don't rely on the test name?

**Review by zzacharo — APPROVED** [review]

File: invenio_rdm_records/services/review/service.py [hunk 6]
```diff
@@ -172,6 +178,17 @@ def submit(self, identity, id_, data=None, require_review=False, uow=None):
         # Check permission
         self.require_permission(identity, "update_draft", record=draft)
 
+        can_submit_record = current_communities.service.config.permission_policy_cls(
+            "submit_record",
+            community_id=draft.parent.review.get_object()
```
Comment (sakshamarora1) [inline invenio_rdm_records/services/review/service.py:183]: Let's extract the community_id separately, to make it more readable.

File: tests/resources/test_resources_communities.py [hunk 7]
```diff
@@ -462,16 +461,17 @@ def test_accept_public_record_in_restricted_community(
...
+    response = client.post(
+        f"/requests/{request_id}/actions/accept",
+        headers=headers,
+        json={},
+    )
+    assert response.status_code == 400
+    assert (
+        response.json["message"]
+        == "A public record cannot be included in a restricted community."
+    )
```
Comment (jrcastro2) [inline tests/resources/test_resources_communities.py:474]: Major: Not sure if this is a leftover, but the behaviour hasn't changed and tests fail with this change. I will revert it and we can check this later @anikachurilova

Comment (jrcastro2) [reply tests/resources/test_resources_communities.py:474]: Discussed IRL, there are changes needed due to the new permission

**Pull request merged**